### DISCUSSION TITLE
Fetch All API's of Umami

### DISCUSSION THREAD

**Discussion by Prannesh45** [umami-software/umami#1969] [Q&A]
Hi Guys, is there any way to fetch all API's of umami as json ?

**Comment by makingtheimpact** [comment]
I was wondering if API was available. I just found this [https://umami.is/docs/api](https://umami.is/docs/api) but I don't know how to set it up yet.

**Comment by briancao** [comment]
We're thinking of adding swagger at some point, but its low priority at this moment.

There is a typescript library, [Umami API Client](https://umami.is/docs/api-client) that you might be interested in.  It has most of our end-points mapped out to specific functions that are typed.
